http-client/src/main/java/io/micronaut/http/client/filter/HttpClientFilterResolver.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2017-2019 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.client.filter;
+
+import io.micronaut.context.annotation.BootstrapContextCompatible;
+import io.micronaut.context.annotation.Parameter;
+import io.micronaut.context.annotation.Prototype;
+import io.micronaut.core.annotation.AnnotationMetadata;
+import io.micronaut.core.annotation.AnnotationMetadataResolver;
+import io.micronaut.core.annotation.AnnotationValue;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.util.ArrayUtils;
+import io.micronaut.core.util.PathMatcher;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.core.util.Toggleable;
+import io.micronaut.http.HttpMethod;
+import io.micronaut.http.HttpRequest;
+import io.micronaut.http.annotation.Filter;
+import io.micronaut.http.annotation.HttpFilterStereotype;
+import io.micronaut.http.filter.HttpClientFilter;
+import io.micronaut.http.filter.HttpFilterResolver;
+
+import javax.annotation.Nullable;
+import java.util.*;
+
+/**
+ * Resolves filters for http clients.
+ *
+ * @author James Kleeh
+ * @since 1.3.0
+ */
+@Internal
+@Prototype
+@BootstrapContextCompatible
+public class HttpClientFilterResolver implements HttpFilterResolver {
+
+    private final AnnotationMetadataResolver annotationMetadataResolver;
+    private final List<HttpClientFilter> clientFilters;
+    private final Collection<String> clientIdentifiers;
+    private final AnnotationValue annotationValue;
+
+    /**
+     * Default constructor.
+     *
+     * @param clientIdentifiers          The client identifiers
+     * @param annotationValue            The filter annotation
+     * @param annotationMetadataResolver The annotation metadata resolver
+     * @param clientFilters              All client filters
+     */
+    public HttpClientFilterResolver(
+            @Parameter @Nullable Collection<String> clientIdentifiers,
+            @Parameter @Nullable AnnotationValue annotationValue,
```

Comment:
There might be multiple filters with customer `stereotype` annotations. Taking into account we are passing information about just one annotation into resolver, is it correct to assume Micronaut will not support a case when multiple client annotations are applied on a client? 

For example with a current implementation apparently it won't be possible to do something like
`@Metered @OauthClientCredentials @Client() SomeClient client`, correct?

Reply:
@svishnyakoff Currently that is correct. If you'd like to create a separate issue for the enhancement of the feature that would be OK